lib/logstash/inputs/http_poller.rb

```diff
@@ -64,7 +67,57 @@ def register
 
     @logger.info("Registering http_poller Input", :type => @type,
                  :urls => @urls, :interval => @interval, :timeout => @timeout)
-  end # def register
+
+    setup_requests!
+  end
+
+  private
+  def setup_requests!
+    @requests = Hash[@urls.map {|name, url| [name, normalize_request(url)] }]
+  end
+
+  private
+  def normalize_request(url_or_spec)
+    if url_or_spec.is_a?(String)
+      res = [:get, url_or_spec]
+    elsif url_or_spec.is_a?(Hash)
+      # The client will expect keys / values
+      spec = Hash[url_or_spec.clone.map {|k,v| [k.to_sym, v] }] # symbolize keys
+
+      # method and url aren't really part of the options, so we pull them out
+      method = (spec.delete(:method) || :get).to_sym.downcase
+      url = spec.delete(:url)
+
+      # We need these strings to be keywords!
+      spec[:auth] = {user: spec[:auth]["user"], pass: spec[:auth]["pass"]} if spec[:auth]
```

Comment:
@jordansissel I'm actually thinking of getting rid of that option simply because you can also set the password as part of the URL with interpolation. What are your thoughts?